documentation/contributing/maintainer_pr/index.markdown

```diff
@@ -5,65 +5,84 @@ slug: maintainer_pr
+### Exceptions to Two Approval Policy
 
-Contributors should indicate their desired merge-policy in the main PR comment.
+MoveIt Maintainers only need need one other review if the following criteria are met:
+- The review is from another maintainer
+- The change is not too large or controversial
+- All feedback has been addressed by the requestor
 
-### Feature Branches in Upstream Repositories
+Once this criteria is met, the maintainer who is the original author is free to merge their own PR in.
 
-Pull requests should originate from a user's fork of the main MoveIt upstream repository as much as possible. Github has a feature that allows a MoveIt maintainer to contribute to a contributor's pull request branch [even if the branch is on another user's fork](https://help.github.com/articles/committing-changes-to-a-pull-request-branch-created-from-a-fork/). Use of this is encouraged.
+Other exceptions for only one approval required:
+- A very small or trivial fix
+- Changes to ``moveit_experimental`` and ``moveit_tutorials`` packages
 
-In rare cases, if a maintainer expects support in the development of a patch from non-maintainer users, they are free to create a new branch in the upstream repository instead. This enables other maintainers to directly push changes there and enables users to add pull-requests targeting the feature branch. To keep the list of branches clear and unambiguous, **names for such branches should always follow the pattern ``pr-<ros distribution>-<keyword description>``**. This makes it clear these branches are not relevant to users not involved in the respective request.
+## GitHub Merge Policies & Best Practices
+
+GitHub provides [several merge policies ](https://help.github.com/articles/about-merge-methods-on-github). To avoid clutter in our commit history, **usually pull requests should be `squash-merged`** (Github supports this via a drop-down list on the "Merge" button), particularly if there are only single commits or several "fixup commits".
+
+However, for larger changes to the code, it's absolutely desired to organize the pull request into multiple, clearly separated, but interrelated commits. In such cases, it is usually also meaningful to keep this commit history, thus being able to track these individual commits also later in the future. In this case, a maintainer should ask the contributor to cleanup "fixup" commits via squashing or interactive rebasing before eventually performing a proper `merge commit`.
 
-### The Latest Branch Preserves All Original Commits
+Multiple, independent commits should not show up in a single PR, but should be split into multiple independent PRs. GitHub's `rebase-merge` policy, which could be used in such cases, unfortunately drops any notion of the related pull request id. To summarize:
+
+- Single commits -> squash
+- Multiple fixup-style commits -> squash
+- Multiple **clean**, interrelated commits -> merge commit
+- Multiple **clean**, independent commits -> rebase and merge
```

Comment:
Fully independent commits shouldn't be in a single PR, so this case almost never applies. If they are grouped for a reason (i.e. clang-tidy fixups), a merge commit with a meaningful title is better imo.

Reply:
Note I did not write this section, I only capitalized the first letter.

I'm guessing there is occasionally a use case for this, and its worth keeping this in here.